assets/app/components/site/Pages/pagesContainer.js

```diff
@@ -14,85 +16,65 @@ class Pages extends React.Component {
   }
 
   componentDidMount() {
-    siteActions.fetchFiles(this.props.site, this.getPath(this.props.params));
+    fetchFiles(this.props.site, this.props.params);
   }
 
   componentWillReceiveProps(nextProps) {
     const { params, site } = nextProps;
-    const nextFiles = this.getFilesByPath(site.files, this.getPath(params));
-    const files = this.getFilesByPath(site.files, this.getPath(params));
+    const nextFiles = getFilesByPath(site.files, getPath(params));
+    const files = getFilesByPath(site.files, getPath(params));
 
     if (files.length && nextFiles.length === files.length) return;
 
-    siteActions.fetchFiles(site, this.getPath(params));
-  }
-
-  getLinkFor(page, id, branch) {
-    const path = page.path;
-
-    return this.isDir(page) ?
-      `/sites/${id}/tree/${path}` : `/sites/${id}/edit/${branch}/${path}`;
+    fetchFiles(site, params);
   }
 
-  getPath(routeParams) {
-    const { splat, fileName } = routeParams;
-    let path = '';
+  render() {
+    const { site, params } = this.props;
 
-    if (splat) {
-      path = `${splat}/${fileName}`;
+    if (!site) {
+      return null;
     }
-    else if (fileName) {
-      path = fileName;
+
+    if (hasNavigationJsonContent(site)) {
+      return <NavigationJsonPageList site={ site } />;
     }
 
-    return path;
+    const path = getPath(params);
+    const files = getFilesByPath(site.files, path) || [];
+    return <PageList site={ site } files={ files }/>;
   }
+}
 
+const fetchFiles = (site, params) => {
+  siteActions.fetchFiles(site, getPath(params));
+};
 
-  getFilesByPath(files = [], startingPath = '') {
-    const isRoot = (startingPath === '');
-    const path = (!isRoot) ? `${startingPath}/` : '((?!/).)*$';
-    const startsWithPath = new RegExp(`^${path}`);
+const getPath = (routeParams) => {
```

Comment:
Where is the logical line that makes the most sense today? What responsibility do we want the actions to have? In this case, I see `siteActions.fetchFiles(site, path)` and it makes sense to me, much more sense than `fetchFiles(site, hopeYouGetWhatYouWantInThisHash)`.

I guess what I'm getting at is where should this logic live? Why are we taking a generic `params` object and transform it into something we can actually use-- we don't do anything else with the `params` object.

That tells me we should change our interface (someday) in pagesContainer. It never ever ever ever really wants a `params` object. It really wants whatever it gets back from `getPath(params)` as that's what it does to transform `params` for use every single time.

In my perfect world, the well-formed, logically coherent path is passed to the pagesContainer instead of the `params` object. This path could be computed when the redux state is set and we'd never have to think about how the value is actually encoded or decode the form ever again in our view (or the action for that matter). That's where I'd want the logic to live.